fixtures/parallelizer/__init__.py

```diff
@@ -245,41 +245,59 @@ def __init__(self, config):
                 date=self.config.option.sprout_date,
                 lease_time=self.config.option.sprout_timeout
             )
-            self.terminal.write("Pool {}. Waiting for fulfillment ...\n".format(pool_id))
+            self.println("Pool {}. Waiting for fulfillment ...".format(pool_id))
             self.sprout_pool = pool_id
             at_exit(self.sprout_client.destroy_pool, self.sprout_pool)
             if self.config.option.sprout_desc is not None:
                 self.sprout_client.set_pool_description(
                     pool_id, str(self.config.option.sprout_desc))
+
+            def detailed_check():
+                try:
+                    result = self.sprout_client.request_check(self.sprout_pool)
+                except SproutException as e:
+                    # TODO: ensure we only exit this way on sprout usage
+                    try:
+                        self.sprout_client.destroy_pool(pool_id)
+                    except:
+                        pass
```

Comment:
you can use the `with diaper:` in here, a standardized usage of the diaper pattern :D